Tools/Algorithms/parser.py

```diff
@@ -0,0 +1,31 @@
+def parse_input(input_file):
+    """
+    Parse WarpX input file.
+
+    Parameters
+    ----------
+    input_file : string
+        Path to input file.
+
+    Returns
+    -------
+    input_dict : dictionary
+        Dictionary storing WarpX input parameters
+        (parameter's name stored as key, parameter's value stored as value).
+    """
+    input_dict = dict()
+    for line in open(input_file):
+        sline = line.split('=')
+        # skip lines that are commented out or blank
+        skip_line = sline[0].startswith('#') or sline[0].startswith('\n')
```

Comment:
The parameter values can extend over multiple lines (for instance function expressions), so there may be cases where `len(sline) == 1` when it is not a comment nor blank line, which would cause an error below when using `sline[1]`.

Reply:
About this, it's a good point. For now I added an extra condition that checks whether the line has only one entry, in which case it's skipped, but I still have to figure out how to possibly add that entry to a previous key of the dictionary, assuming it's always a continuation of a previous input parameter. It is not relevant for the purpose of this PR, but in general it is something that the parser should be able to do correctly.